force-app/main/default/classes/MetaDataUtilityCls.cls

```diff
@@ -1,292 +1,328 @@
+                {
+                    Schema.SObjectType relatedObjType = selectedField.getReferenceTo()[0];
+                    relatedLookupMap.put(relatedObjType.getDescribe().getName(), relatedObjType.getDescribe().getLabel());
+                    lookUpAPINameMap.put(selectedField.getLabel(), selectedField.getName());
+                }
             }
         }
-        System.debug('relatedlookupMap: ' + relatedlookupMap);
-        return relatedlookupMap;
     }
-
+    
     //This method is used to get the fields from the object
-    @AuraEnabled(Cacheable=true)
-    public static Map<String, String> getSObjectFields(String objectName){
-        Map<String, String> sObjectFieldsMap = new Map<String, String>();
+    @AuraEnabled( Cacheable=true )
+    public static Map<String, String> getSObjectFields(String objectName)
+    {
         Map<String, String> fieldSetMap = new Map<String, String>();
-        SObjectType sobjType = Schema.getGlobalDescribe().get(objectName);
-        Map<String, Schema.SObjectField> fieldsMap = sobjType.getDescribe().fields.getMap();
-        for(Schema.SObjectField field : fieldsMap.values()){
-            Schema.DescribeFieldResult fieldResult = field.getDescribe();
-            Schema.DisplayType fieldType = fieldResult.getType();
-
-            if(fieldResult.isAccessible()){
-                sObjectFieldsMap.put(fieldResult.getName(),fieldResult.getLabel());
-                fieldNameAPISetMap.put(fieldResult.getLabel(), fieldResult.getName());
+        Map<String, String> fieldNameAPISetMap = new Map<String, String>();
+        Map<String, String> sObjectFieldsMap = new Map<String, String>();
+        Map<String, Schema.SObjectField> fieldsMap = Schema.getGlobalDescribe().get(objectName)?.getDescribe().fields.getMap();
+        
+        if(fieldsMap != null)
+        {
+            for (Schema.SObjectField field : fieldsMap.values())
+            {
+                Schema.DescribeFieldResult fieldResult = field.getDescribe();
+                if (fieldResult.isAccessible())
+                {
+                    sObjectFieldsMap.put(fieldResult.getName(), fieldResult.getLabel());
+                    fieldNameAPISetMap.put(fieldResult.getLabel(), fieldResult.getName());
+                }
+            }
+            
+            for (String key : getSortedKeySet(sObjectFieldsMap))
+            {
+                fieldSetMap.put(key, sObjectFieldsMap.get(key));
             }
         }
-        for(String key:getSortedKeySet(sObjectFieldsMap)){
-            fieldSetMap.put(key, sObjectFieldsMap.get(key));
-        }
+        
+        
         return fieldSetMap;
     }
-
+    
     //This method is used to get all the objects from the org
-    @AuraEnabled(Cacheable=true)
-    public static Map<String, String> getAllSObjectsInOrg(){
+    @AuraEnabled( Cacheable=true )
+    public static Map<String, String> getAllSObjectsInOrg()
```

Comment:
@michaelmuse I am curious of your thoughts on thevchanges getAllsObjects in this method. I added a few more criteria and and it really brings the total from 644 objects to 209. I think that is for the better. It removes objects like "ApexClass" which I cant see anyone ever needing a listener for.